pkg/daemon/writer.go

```diff
@@ -207,7 +213,7 @@ func (w *NodeStateStatusWriter) getNodeState() (*sriovnetworkv1.SriovNetworkNode
 	var lastErr error
 	var n *sriovnetworkv1.SriovNetworkNodeState
 	err := wait.PollImmediate(10*time.Second, 5*time.Minute, func() (bool, error) {
-		n, lastErr = w.client.SriovnetworkV1().SriovNetworkNodeStates(vars.Namespace).Get(context.Background(), vars.NodeName, metav1.GetOptions{})
+		n, lastErr = w.client.SriovnetworkV1().SriovNetworkNodeStates(vars.Namespace).Get(context.Background(), vars.NodeName, metav1.GetOptions{ResourceVersion: "0"})
```

Comment:
I always keep forgetting can you please remind me why we need the `ResourceVersion` to be zero here?

Reply:
from k8s docs[1]:


_Return data at any resource version. The newest available resource version is preferred, but strong consistency is not required; data at any resource version may be served. It is possible for the request to return data at a much older resource version that the client has previously observed, particularly in high availability configurations, due to partitions or stale caches. Clients that cannot tolerate this should not use this semantic._

[1] https://kubernetes.io/docs/reference/using-api/api-concepts/#resourceversion-in-metadata

Reply:
If you put ResourceVersion="0" you get a object from the kube-apiserver cache.
If you request latest version (without setting ResourceVersion) the request always needs to reach etcd and retrieve the latest version.

I was running this code at 6k nodes scale. It refreshes the state every 30s. This means that there was a constant 200 QPS that needs to fetch the object from the etcd and hardly ever (if ever) change anything.

In general the preferable way to keep track of the object is to use a Watcher:
ListAndWatch handles the whole flow but in general:
- List objects with RV=0 to get latest state from the kube-apiserver cache
- Watch starting from the last version so you get all the updates since the returned state.

I'm not aware of any usecase in which you really need to have latest state from the etcd.